Assets/Scripts/NPCs/BrutusResponder.cs

```diff
@@ -0,0 +1,219 @@
+﻿using System;
+using NPCs.Components;
+using UnityEngine;
+
+public enum BrutusResponderStates
+{
+    Chasing,
+    Responding,
+    Searching,
+    Patrolling,
+    Rotating,
+    Waiting,
+}
+
+namespace NPCs
+{
+    [RequireComponent(typeof(Chaser))]
+    [RequireComponent(typeof(Responder))]
+    [RequireComponent(typeof(Searcher))]
+    [RequireComponent(typeof(Patroller))]
+    [RequireComponent(typeof(Waiter))]
+    [RequireComponent(typeof(Enemy))]
+    public class BrutusResponder : BaseStateAi<BrutusResponderStates>
+    {
+        [HideInInspector] public Chaser chaser;
+        [HideInInspector] public Responder responder;
+        [HideInInspector] public Searcher searcher;
+        [HideInInspector] public Patroller patroller;
+        [HideInInspector] public Waiter waiter;
+        [HideInInspector] public Enemy enemy;
+
+        public int numSearches = 3;
+        public GameObject questionMark;
+        private Animator animator;
+        private int curNumSearches = 0;
+
+        protected override void Awake()
+        {
+            base.Awake();
+
+            chaser = Utils.GetRequiredComponent<Chaser>(this);
+            responder = Utils.GetRequiredComponent<Responder>(this);
+            searcher = Utils.GetRequiredComponent<Searcher>(this);
+            patroller = Utils.GetRequiredComponent<Patroller>(this);
+            waiter = Utils.GetRequiredComponent<Waiter>(this);
+            // animator = Utils.GetRequiredComponentInChildren<Animator>(this);
+            enemy = Utils.GetRequiredComponent<Enemy>(this);
+        }
+
+        public void Start()
+        {
+            chaser.OnSeePlayer += () => SetState(BrutusResponderStates.Chasing);
+            chaser.OnLosePlayer += () => nextState = BrutusResponderStates.Searching; SetState(BrutusResponderStates.Waiting);
```

Comment:
Loose `SetState(BrutusResponderStates.Waiting);` at the end there?

Reply:
Yep that's supposed to be there, let me fix the formatting though so it's actually legible